src/ip_notifier.cpp

```diff
@@ -311,13 +311,60 @@ struct ip_change_notifier_impl final : ip_change_notifier
 #endif // TORRENT_USE_SC_NETWORK_REACHABILITY
 
 #elif defined TORRENT_WINDOWS
-	// TODO: ip_change_notifier_win
-#else
-	// TODO: ip_change_notifier_default
+struct ip_change_notifier_impl final : ip_change_notifier
+{
+	explicit ip_change_notifier_impl(io_service& ios)
+		: m_hnd(ios, WSACreateEvent())
+	{
+		if (!m_hnd.is_open()) aux::throw_ex<system_error>(WSAGetLastError(), system_category());
+		m_ovl.hEvent = m_hnd.native_handle();
+	}
+
+	// noncopyable
+	ip_change_notifier_impl(ip_change_notifier_impl const&) = delete;
+	ip_change_notifier_impl& operator=(ip_change_notifier_impl const&) = delete;
+
+	~ip_change_notifier_impl() override
+	{
+		cancel();
+		// the reason to call close() here is because the
+		// object_handle destructor doesn't close the handle
+		m_hnd.close();
```

Comment:
I'm a bit surprised the destructor doesn't close the handle. That means its life time isn't managed by anything. If any code are introduced in the constructor that may throw, the handle will leak. Currently the only thing that can throw in there is failing to create the event itself.